### PR TITLE
shell: set timezone implementation

### PR DESCRIPTION
**Pull request by fridex**
Based on https://github.com/cockpit-project/cockpit/issues/632

### CODE REVIEW

File: pkg/shell/cockpit-server.js [hunk 1]
```diff
@@ -683,9 +683,43 @@ PageSystemInformationChangeSystime.prototype = {
...
+                    value: timezones[j],
+                    text: timezones[j],
+                    selected: timezones[j] == curr_timezone
+                }));
+            }
+
+            $('#systime-timezones').selectpicker('refresh');
+        }
+
+        cockpit.file("/usr/share/zoneinfo/zone.tab").read()
```
Comment (stefwalter) [inline pkg/shell/cockpit-server.js]: I would suggest using this ```timedatectl list-timezones```

**Comment by andreasn** [comment]
It seems like all the places are repeated 3 or 4 times over in the list.

**Comment by andreasn** [comment]
It seems we need to make some relayout to the time and date dialogs to add some descriptive labels.

The biggest issue with the dropdown is the amount of items in it, so it will be slow to navigate. Therefore I suggest to use a combobox and make it possible to filter down for the locations using that.
The dropdown also needs to be limited in height so that the list remains scannable, I've made it 10 items in the mockup.

The names should be formatted with spaces instead of undescores, eg. Addis Ababa instead of Addis_Ababa.

Mockup

**Comment by fridex** [comment]
> It seems like all the places are repeated 3 or 4 times over in the list.

Strange. Are they repeated in "timedatectl list-timezones"?

**Comment by andreasn** [comment]
Upon first launch I get a [x] button that I need to hit before I can change away from the incorrect time zone "New York" to the correct "Stockholm"
Why not just a dropdown here?

screenshot from 2015-05-29 13-01-46

**Comment by andreasn** [comment]
The dropdown itself behaves a bit odd:

screenshot from 2015-05-29 13-00-10

Might just as well make the dropdown full width (to make it consistent with dropdowns elsewhere in Cockpit)

**Comment by andreasn** [comment]
screenshot from 2015-05-29 12-56-58

The Change button doesn't line up with the rest of the elements. I think this might be because the footer div is inside the main dialog div somehow

**Comment by andreasn** [comment]
the width of the dialog unfortunatley make the date, but especially the time boxes super wide. They are more efficient if they can indicate about how many letters or numbers are supposed to be inputed in them. So time should "fit" about 2 numbers.

**Comment by andreasn** [comment]
In worst case we can give the dialog a custom size, invent a custom modal-md class with a width of 400px

**Comment by fridex** [comment]
> Upon first launch I get a [x] button that I need to hit before I can change away from the incorrect time zone "New York" to the correct "Stockholm"
> Why not just a dropdown here?
> ...
> The dropdown itself behaves a bit odd

I used available combobox. I didn't change anything in styling.

> the width of the dialog unfortunatley make the date, but especially the time boxes super wide

I added new dialog. I think this might be better option, otherwise there is a lot of white spacing.

**Comment by andreasn** [comment]
> It seems like all the places are repeated 3 or 4 times over in the list.

I figured out how to reproduce this one.
 * Open the dialog.
 * Search for "Stockholm"
 * Close the dialog
 * Open the dialog
 * Search for "Stockholm"
 * Close the dialog
 * Open the dialog
 * Search for "Stockholm"

Stockholm will now appear 3 times in the list. Close and open to have it appear a 4th time in the list.

**Comment by andreasn** [comment]
I realized making the dropdown the proper width requires a new version of Patternfly
https://github.com/patternfly/patternfly/issues/60

**Comment by andreasn** [comment]
https://github.com/patternfly/patternfly/issues/61 is also related to this issue

**Comment by andreasn** [comment]
> I figured out how to reproduce this one.

This happens also when you set the time btw

**Comment by fridex** [comment]
> Stockholm will now appear 3 times in the list.

Thanks, should be fixed now.

**Comment by andreasn** [comment]
When changing cities a lot, it sometimes tends to get stuck on some of them, ie. Set city, open dialog again, set another city, open again, first city is still there.
How much did you test this?

**Comment by andreasn** [comment]
Screencast of the issue above (note that I tweaked the css a bit locally)
https://cloud.gnome.org/index.php/s/eWphmja8LfH6N0l

**Comment by fridex** [comment]
It looks like old fix was not good. I tested it; it should work now.

**Comment by andreasn** [comment]
Updated patternfly version that fixes some of the issues is in https://github.com/cockpit-project/cockpit/pull/2408

**Comment by andreasn** [comment]
Tested and works great now!

**Pull request closed without merging**